ip/event_manager.go

```diff
@@ -131,7 +137,8 @@ func (m *manager) waitTwiceLeaseTimeOrReallocation(ctx context.Context) {
 // otherwise. It can happen if the current manager stopping has been cancelled.
 func (m *manager) stopOrder(ctx context.Context) bool {
 	m.stopMutex.RLock()
-	defer m.stopMutex.RUnlock()
+	stopper := m.stopper
+	m.stopMutex.RUnlock()
```

Comment:
I'm not super confident about this change in the mutex unlocking. What happens if the stopping is canceled between line 147 and 151?